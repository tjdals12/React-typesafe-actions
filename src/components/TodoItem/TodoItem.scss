@import 'utils';

.todo-item {
    display: flex;
    align-items: center;
    padding: 1rem;

    .input {
        margin-right: 1rem;
    }

    .text {
        flex: 1;
        font-family: 'Sunflower', sans-serif;
        font-size: 1rem;
        font-weight: 300;
        letter-spacing: 0.2rem;

        word-break: break-all;

        cursor: pointer;

        &.done {
            text-decoration: line-through;
        }
    }

    .remove {
        cursor: pointer;
        color: $oc-red-6;

        &:hover {
            color: $oc-red-4;
        }

        &:active {
            color: $oc-red-9;
        }
    }

    &:nth-child(odd) {
        background: $oc-gray-1;
    }

    & + & {
        border-top: 1px solid $oc-gray-3;
    }
}
