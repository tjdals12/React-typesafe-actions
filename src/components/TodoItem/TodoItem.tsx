import React from 'react';
import classNames from 'classnames/bind';
import useTodoActions from 'hooks/useTodoActions';
import styles from './TodoItem.scss';

const cx = classNames.bind(styles);

type TodoItemProps = {
    todo: { id: number; text: string; done: boolean };
};

export default function TodoItem({ todo }: TodoItemProps): JSX.Element {
    const { toggle, remove } = useTodoActions(todo.id);

    return (
        <li className={cx('todo-item')}>
            <input className={cx(`input ${todo.done ? 'done' : ''}`)} type="checkbox" checked={todo.done} readOnly />
            <span className={cx(`text ${todo.done ? 'done' : ''}`)} onClick={toggle}>
                {todo.text}
            </span>
            <span className={cx('remove')} onClick={remove}>
                X
            </span>
        </li>
    );
}
