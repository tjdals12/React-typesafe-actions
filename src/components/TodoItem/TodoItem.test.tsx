import React from 'react';
import { render } from '@testing-library/react';
import TodoItem from './TodoItem';

describe('<TodoItem />', () => {
    it('matches snapshot', () => {
        const utils = render(<TodoItem todo={{ id: 0, text: 'Something 1', done: false }} />);
        expect(utils.container).toMatchSnapshot();
    });

    it('has a input and two spans', () => {
        const utils = render(<TodoItem todo={{ id: 0, text: 'Something 1', done: false }} />);
        expect(utils.container.querySelector('.input')).not.toBeNull();
        expect(utils.container.querySelector('.text')).not.toBeNull();
        expect(utils.container.querySelector('.remove')).not.toBeNull();
    });
});
