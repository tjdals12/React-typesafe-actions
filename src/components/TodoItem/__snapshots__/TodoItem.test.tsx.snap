// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItem /> matches snapshot 1`] = `
<div>
  <li
    class="todo-item"
  >
    <input
      class="input "
      readonly=""
      type="checkbox"
    />
    <span
      class="text"
    >
      Something 1
    </span>
    <span
      class="remove"
    >
      X
    </span>
  </li>
</div>
`;
